extensions/json/src/main/java/io/deephaven/json/ValueOptionsRestrictedUniverseBase.java

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json;
+
+import org.immutables.value.Value.Check;
+
+import java.util.EnumSet;
+
+/**
+ * A base {@link ValueOptions} where the implementation has a clearly defined {@link #universe()}.
+ */
+public abstract class ValueOptionsRestrictedUniverseBase extends ValueOptions {
+
+    /**
+     * The allowed types. Must be a subset of {@link #universe()}.
+     */
+    @Override
+    public abstract EnumSet<JsonValueTypes> allowedTypes();
+
+    /**
+     * The universe of possible allowed types.
+     */
+    public abstract EnumSet<JsonValueTypes> universe();
+
+    @Check
+    void checkAllowedTypes() {
+        for (JsonValueTypes type : allowedTypes()) {
```

Comment:
`universe.containsAll(allowedTypes()`
Include universe and allowed types in message.